File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int change;       // the given amount of change
int coins_needed; // the minimum coins needed to make the given amount of change

void use_coin(int coin_type)
{
    while (change >= coin_type)
    {
        coins_needed++;
        change -= coin_type;
    }
}

int main(void)
{
    do
    {
        change = get_int("Change owed: ");
    }
    while (change <= 0);

    use_coin(25);
    use_coin(10);
    use_coin(05);
    use_coin(01);

    printf("%i\n", coins_needed);
}

File: pset6/sentimental-cash/cash.py
from cs50 import get_float

# Keep prompting user until we get a non-negative number
while True:
    dollars = get_float("Change: ")
    if dollars >= 0:
        break

# Convert from dollar to cent
cents = int(dollars * 100)

# The minimum coins needed
coins = 0

# Take coins from the largest to the smallest
for value in [25, 10, 5, 1]:
    while cents >= value:
        coins += 1
        cents -= value

# Given the money back
print(coins)
